crates/biome_js_analyze/tests/specs/correctness/noUnusedVariables/validScriptGlobals.ts

```diff
@@ -0,0 +1,8 @@
+/* should not generate diagnostics */
+interface Array<T> {
+	g: (a: T) => void;
+}
+
+namespace M {
+	export const n = 1;
+}
```

Comment:
I would leave a brief comment to explain _why_ this is a valid case, because this issue is a bit convoluted.